controllers/utils/etcd.go

```diff
@@ -33,6 +36,30 @@ func IsEtcdDisruptionAllowed(ctx context.Context, cl client.Client, log logr.Log
 		log.Info("Etcd disruption is allowed")
 		return true, nil
 	}
-	log.Info("Etcd disruption is not allowed")
+
+	// No disruptions allowed, so the only case we should remediate is that the node in question is already one of the disrupted ones
+	// The PDB doesn't disclose which node is disrupted
+	// So we have to check the etcd guard pods
+	podList := &corev1.PodList{}
+	if err := cl.List(ctx, podList, &client.ListOptions{
+		Namespace:     etcdNamespace,
+		LabelSelector: labels.SelectorFromSet(pdb.Labels),
+	}); err != nil {
+		return false, err
+	}
+	for _, pod := range podList.Items {
+		if pod.Spec.NodeName == node.Name {
+			for _, condition := range pod.Status.Conditions {
+				if condition.Type == corev1.PodReady && condition.Status == corev1.ConditionFalse {
+					log.Info("Etcd disruption not allowed, but the node is already disrupted, so remediation is allowed")
+					return true, nil
+				}
+			}
+			log.Info("Etcd disruption not allowed, but the node is already disrupted, so remediation is allowed")
+			return true, nil
+		}
+	}
```

Comment:
We are looking for the etcd guard pods of the unhealthy control plane node.
Wouldn't it be easier to check for pod `etcd-guard-`NODENAME (e.g., `etcd-guard-master-0`) at `openshift-etcd` namespace instead?

Reply:
Names can change... 🤷🏼‍♂️

Reply:
Indeed, this check is more bulletproof